### PR TITLE
ModCloud implementation and Example Mod examples

### PR DESCRIPTION
**Pull request by ArctanDeveloper**
### What is the new feature?
Moddable clouds.
### Why should this be part of tModLoader?
It's extra abilities for modders, and avoids them having to do ILEdits for a simple feature.
### Are there alternative designs?
Possibly, but I'm not aware of them.
### Sample usage for the new feature
Making a custom cloud.
### ExampleMod updates
A simple ExampleCloud that is manually loaded and an AdvancedExampleCloud showing off how to change spawn params.

dotnet_2024-09-20_16-35-27

### CODE REVIEW

File: ExampleMod/Assets/Textures/CloudsT/SimpleCloud.png
Comment (JavidPack) [inline ExampleMod/Assets/Textures/CloudsT/SimpleCloud.png:1]: Why is this here?

File: ExampleMod/ExampleMod.cs [hunk 1]
```diff
@@ -13,6 +13,7 @@ public partial class ExampleMod : Mod
 
 		public override void Load() {
 			// Registers a new custom currency
+			CloudLoader.AddCloudFromTexture<SimpleModCloud>(this, AssetPath + "Textures/CloudsT/SimpleCloud");
```
Comment (JavidPack) [inline ExampleMod/ExampleMod.cs]: Move this down so the comment above still comments the ExampleCustomCurrencyId  line

File: patches/tModLoader/Terraria/ModLoader/ModContent.cs [hunk 2]
```diff
@@ -462,6 +467,7 @@ private static void SetupRecipes(CancellationToken token)
 	internal static void UnloadModContent()
 	{
 		MenuLoader.Unload(); //do this early, so modded menus won't be active when unloaded
+		CloudLoader.Unload();
```
Comment (JavidPack) [inline patches/tModLoader/Terraria/ModLoader/ModContent.cs:470]: These usually go in ModContent.Unload

File: patches/tModLoader/Terraria/release_extras/LaunchUtils/dotnet-install.ps1
Comment (JavidPack) [inline patches/tModLoader/Terraria/release_extras/LaunchUtils/dotnet-install.ps1:1]: Delete this file

File: patches/tModLoader/Terraria/release_extras/dotnet/.zip
Comment (JavidPack) [inline patches/tModLoader/Terraria/release_extras/dotnet/.zip:1]: Delete this file

File: patches/tModLoader/Terraria/release_extras/tModLoader-Logs/Launch.log
Comment (JavidPack) [inline patches/tModLoader/Terraria/release_extras/tModLoader-Logs/Launch.log:1]: Delete this file

File: patches/tModLoader/Terraria/release_extras/tModLoader-Logs/Natives.log
Comment (JavidPack) [inline patches/tModLoader/Terraria/release_extras/tModLoader-Logs/Natives.log:1]: Delete this file

**Review by JavidPack — CHANGES_REQUESTED** [review]
I noticed some simple issues with the PR to fix. I haven't tested it yet directly.

File: patches/tModLoader/Terraria/ModLoader/ModContent.cs (hunk 2 above)
Comment (ArctanDeveloper) [reply patches/tModLoader/Terraria/ModLoader/ModContent.cs:470]: Clouds are present on the menu, if you don't unload early it crashes.

File: ExampleMod/Assets/Textures/CloudsT/SimpleCloud.png
Comment (ArctanDeveloper) [reply ExampleMod/Assets/Textures/CloudsT/SimpleCloud.png:1]: To show off manual showcase of SimpleCloud registration

**Comment by JavidPack** [comment]
Thanks a bunch.
